build_tools/cmake/setup_sccache.sh

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# ccache (https://ccache.dev/) setup, with read/write + local/remote options.
+#
+# Defaults to only reading from the shared remote cache (hosted on GCS) used by
+# our Linux CI. The postsubmit CI writes to the cache, for presubmit CI and
+# local builds to read from.
+#
+# Local caches can also be used to interface with external remote caches
+# (like https://github.com/actions/cache) by
+#   1. downloading the cache directory
+#   2. sourcing with IREE_READ_LOCAL_SCCACHE=1 IREE_WRITE_LOCAL_SCCACHE=[0,1]
+#   3. building with CMake
+#   4. uploading the cache directory (if writing)
+#
+# Note: this file must be *sourced* not executed.
+
+set -euo pipefail
+
+# Configuration environment variables.
+IREE_READ_REMOTE_SCCACHE="${IREE_READ_REMOTE_SCCACHE:-1}"
+IREE_WRITE_REMOTE_SCCACHE="${IREE_WRITE_REMOTE_SCCACHE:-0}"
```

Comment:
I had a simpler version of this script at https://github.com/iree-org/iree/pull/18499.

With our current Azure sccache setup, I don't think we can use the full set of options here, so I was leaning towards starting simple.